ludwig/data/dataset/ray.py

```diff
@@ -342,6 +357,31 @@ def _prepare_batch(self, batch: pd.DataFrame) -> Dict[str, np.ndarray]:
                 res[c] = res[c].reshape((-1, *reshape))
         return res
 
+    def _augment_batch_fn(self):
+        augmentation_pipeline = self.augmentation_pipeline
+
+        def augment_batch(df: pd.DataFrame) -> pd.DataFrame:
+            if augmentation_pipeline:
+                for c, augmentations in augmentation_pipeline.items():
+                    # TODO: convert to debug message when done with development
+                    logger.info(f"RayDatasetBatcher applying augmentation pipeline to batch for feature {c}")
+
+                    # df[c] is a pd.Series, so we need to convert it to a np.array
+                    # consolidate the individual np.array into a single np.array for the batch
+                    data_to_augment = np.stack(df[c].values, axis=0)
```

Comment:
Nice! You may be able to simplify by using `to_numpy_dataset` and `from_numpy_dataset.

https://github.com/ludwig-ai/ludwig/blob/master/ludwig/utils/dataframe_utils.py#L64-L94

Like:

```
data_to_augment = to_numpy_dataset(df)
augmented_data = {}
for c, augmentations in augmentation_pipeline.items():
    augmented_data[c] = augmentations(torch.tensor(data_to_augment[c])).numpy()
return from_numpy_dataset(augmented_data)    
```

Reply:
This assumes there is a column in the augmentation pipeline for every column in the input df. If not, we could rearrange a little bit to ensure we also include the columns that aren't augmented.

Reply:
done